src/odemis/driver/simcam.py

```diff
@@ -142,6 +143,48 @@ def __init__(self, name, role, image, dependencies=None, daemon=None, blur_facto
         # Convenience event for the user to connect and fire
         self.softwareTrigger = model.Event()
 
+        # show_message(None, 'starting and error wise thread', 'test error', timeout = 1.0)
+        self._errorCreationThread = threading.Thread(target = self._state_error_creation,
+                                                     args = (),
+                                                     name = "Creating and state error")
+        self._errorCreationThread.daemon = True
+        self._errorCreationThread.start()
+
+    def _state_error_creation(self):
+        '''
+        by creating the files 'simcam.HW.error' or 'simcam.HW.good' in the folder '/var/run/odemisd' and hardware
+        error can be produced in the Sim Cam or the error can be 'fixed'
+        After the error is produced the files should be deleted
+        :return:
+        '''
+        def attempt_to_delete_file(del_file):
+            try:
+                os.remove(del_file)
+            except:
+                pass
```

Comment:
Don't try to delete the file, as you'll enter a whole rat nest.
I'd suggest a slightly different behaviour:
* only one single file, /var/run/odemisd/simcam-hw-error`
* if it's not present, the state is "good"
* if it's present, the state is "bad"
* Let the user add/remove the file.

Reply:
That would mean there is now feedback to the user whether or not the function executed correctly. If a user now sees that /var/run/odemisd/simcam-hw-error is not deleted this means that the thread did not execute correctly.
Plus this way ensures that the simulation will not start with a bunch of error's blocking the start up of the simulator in total.

> Because of the try statement the rats in the rats nest maybe won't bite that fast?

Reply:
The function should work fine. Also, the user's feedback is via the .state of the component.